encoder_borsh.go

```diff
@@ -222,18 +251,29 @@ func (enc *Encoder) encodeComplexEnumBorsh(rv reflect.Value) error {
 	if int(enum)+1 >= t.NumField() {
 		return errors.New("complex enum too large")
 	}
+	// Enum is empty
 	field := rv.Field(int(enum) + 1)
-
+	if isTypeBorshEnumEmpty(field.Type()) {
+		return nil
+	}
 	if field.Kind() == reflect.Ptr {
 		field = field.Elem()
 	}
 	if field.Kind() == reflect.Struct {
 		return enc.encodeStructBorsh(field.Type(), field)
 	}
+	// Encode the value if it's a primitive type
+	isPrimitive, err := enc.encodePrimitive(field, nil)
+	if isPrimitive {
+		return nil
+	} else if err != nil {
+		return err
+	}
 	return nil
 }
 
 type BorshEnum uint8
+type BorshEnumEmpty struct{}
```

Comment:
```suggestion
// EmptyVariant is an empty borsh enum variant.
type EmptyVariant struct{}

func (_ *EmptyVariant) UnmarshalWithDecoder(_ *Decoder) error {
	return nil
}

func (_ EmptyVariant) MarshalWithEncoder(_ *Encoder) error {
	return nil
}

```

Reply:
Changed name of type + now using the marshal interface :+1: